### PR TITLE
Supoort sending contents data for Dynamic Template

### PR DESCRIPTION
**Pull request by kazu0620**
Thank you for publishing awsome package!

I tried to use Dynamic Templates feature via this package, but the package didn't have an interface to set the template data. Therefore, I added field of `dynamic_template_data` to Personalization.

## Related official document
https://sendgrid.com/docs/ui/sending-email/how-to-send-an-email-with-dynamic-transactional-templates/#send-a-transactional-email

## Official SDK's implementation in Java 
https://github.com/sendgrid/sendgrid-java/blob/2d82397f32cd14efb7a5fc28fafa019613968f3a/src/main/java/com/sendgrid/helpers/mail/objects/Personalization.java#L36

### CODE REVIEW

File: lib/src/models/email.dart [hunk 1]
```diff
@@ -48,7 +48,7 @@ class Email {
         'from': from.toJson(),
         'reply_to': replyTo?.toJson(),
         'subject': subject,
-        'content': content.map((e) => e.toJson()).toList(),
+        'content': content?.map((e) => e.toJson())?.toList(),
```
Comment (kazu0620) [inline lib/src/models/email.dart:51]: content field is not need to specify at using template.

Comment (dylanshine) [reply lib/src/models/email.dart:51]: This looks good, can you move the content parameter optional in the initializer?

Comment (kazu0620) [reply lib/src/models/email.dart:51]: @dylanshine You're right, I fixed it.

Comment (dylanshine) [reply lib/src/models/email.dart:51]: @kazu0620 Thank you for your contribution

Version 0.1.2 has been published to https://pub.dev/packages?q=sendgrid_mailer

If you wouldn't mind giving it a star on here and pub.dev, to help other people discover the package.

**Pull request merged**